sentry/src/main/java/io/sentry/util/ErrorUtils.java

```diff
@@ -0,0 +1,69 @@
+package io.sentry.util;
+
+import io.sentry.FilterString;
+import io.sentry.SentryEvent;
+import io.sentry.protocol.Message;
+import io.sentry.protocol.SentryException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public final class ErrorUtils {
+
+  /** Checks if an error has been ignored. */
+  @ApiStatus.Internal
+  public static boolean isIgnored(
+      final @Nullable List<FilterString> ignoredErrors, final @NotNull SentryEvent event) {
+    if (event == null || ignoredErrors == null || ignoredErrors.isEmpty()) {
+      return false;
+    }
+
+    final @NotNull Set<String> possibleMessages = new HashSet<>();
+
+    final @Nullable Message eventMessage = event.getMessage();
+    if (eventMessage != null) {
+      final @Nullable String stringMessage = eventMessage.getMessage();
+      if (stringMessage != null) {
+        possibleMessages.add(stringMessage);
+      }
+      final @Nullable String formattedMessage = eventMessage.getFormatted();
+      if (formattedMessage != null) {
+        possibleMessages.add(formattedMessage);
+      }
+    }
+    final @Nullable List<SentryException> exceptions = event.getExceptions();
```

Comment:
AFAIK these are only filled by `EventProcessor`s so they are likely empty when this util is invoked. We should not change the order of filters as that's defined (https://develop.sentry.dev/sdk/telemetry/sessions/#filter-order). So I think it's safe to simply skip looking at these and only rely on the throwable below.